flybywiresim/aircraft
# fix(flight model / bleed air / ice prot): Restore wing anti-ice functionality and simplify bleed air

Fixes #8387

## Summary of Changes

1. Reverts the temporary fix from #8388
1. Removes the `systems_wasm` `engines` aspect and sets bleed air source to auto.
1. Introduces 2 new LVars for engine anti ice push button states.
1. Modifies the `WING ANTI ICE` push button to control their local variables rather than the MSFS aircraft variable directly.
1. Modifies the `ENGINE ANTI ICE` push buttons to control the new local variables rather than the MSFS aircraft variable directly.
1. Modifies the rate that the fuselage deices.
1. Adds an `anti_ice` aspect that monitors the push button LVars and toggles the aircraft variables appropriately.

## Screenshots (if necessary)


N/A

## References




N/A

## Additional context

I have tested this with GSX's de-icing. GSX will toggle on all aircraft anti-ice/de-ice systems in MSFS itself, but will not affect the state of the systems simulation or the overhead push buttons. This means that the next time you turn on the anti ice systems using the overhead push buttons, it won't actually do anything to the MSFS aircraft variables. However, they will then be back in sync, and turning the push buttons off will turn off the MSFS deice.


Discord username (if different from GitHub): fatboyfat_uk

## Testing instructions

This PR makes changes to the bleed air systems as well as re-enabling the anti-ice. The tests below will check that cross-bleed starts still work while testing the anti ice:

1. Spawn somewhere with icing conditions. I've found the easiest way to guarantee this on the ground is to set the custom weather to Snow, move the lowest cloud layer to encompass ground level, then set scatter and density to maximum.
1. Observe the `STRUCTURAL ICE PCT` SimVar somehow - best way is SimvarWatcher or similar. Alternatively, Flow's weather widget shows icing percentage, but it will only drop slowly.
1. Ensure that the variable is slowly increasing i.e. you definitely have icing conditions.
1. Start Engine 1 using APU or ASU.
1. Remove APU or ASU.
1. Perform a cross-bleed start of Engine 2 (following the docs at https://docs.flybywiresim.com/pilots-corner/advanced-guides/engines/crossbleed-start/)
1. Turn on `WING ANTI ICE` with the push button on the overhead panel.
1. While the aircraft is on the ground, this will only work for 30 seconds while the aircraft runs a test. During this time, check that the `STRUCTURAL ICE PCT` SimVar decreases. Once the test is completed, it should start to increase again.
1. When in the air, turn `WING ANTI ICE` back on and check that the `STRUCTURAL ICE PCT` SimVar (eventually) reduces to 0.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
